apps/algorithm/search/Dockerfile

```diff
@@ -2,9 +2,13 @@ FROM python:3.12-slim
 
 WORKDIR /app
 
+RUN apt update && apt install -y curl
```

Comment:
Is there really nothing installed in the python container for making requests? I'm sure you could do a `python -c "import requests......."` or something

Reply:
I did consider that, we can do `python -c "import requests; response = requests.get('link'); exit(0) if response.status_code == 200 else exit(1)"`. I went with this mainly because
1. It's consistent with all the other heathchecks
2. What's one more install? It should be cached by docker and we can pretty much forget about it

I get not wanting to install more stuff, so I can use python if it's preferable

Reply:
Let's use python. As soon as we run `apt` inside a container, build times increase and the image size massively increases (due to the cache & apt lists data now being stored)